src/lib/expansion-panel/expansion-panel-core.ts

```diff
@@ -5,24 +5,50 @@ export interface IExpansionPanelCore {
   open: boolean;
   orientation: ExpansionPanelOrientation;
   animationType: ExpansionPanelAnimationType;
+  trigger: string;
   dispatchToggleEvent(): void;
 }
 
 export class ExpansionPanelCore implements IExpansionPanelCore {
   private _open = false;
   private _orientation: ExpansionPanelOrientation = 'vertical';
   private _animationType: ExpansionPanelAnimationType = 'default';
+  private _trigger = '';
 
   private _clickListener: EventListener = this._onClick.bind(this);
   private _keydownListener: EventListener = this._onKeydown.bind(this);
   private _animationCompleteListener = this._onAnimationComplete.bind(this);
+  private _slotListener = this._handleContentSlotChange.bind(this);
 
   constructor(private _adapter: IExpansionPanelAdapter) {}
 
   public initialize(): void {
     this._adapter.addHeaderListener('click', this._clickListener);
     this._adapter.addHeaderListener('keydown', this._keydownListener);
     this._adapter.setAnimationCompleteListener(this._animationCompleteListener);
+    this._adapter.addContentSlotListener(this._slotListener);
+    this._adapter.setContentId();
+    requestAnimationFrame(() => {
+      this._syncTrigger();
+    });
```

Comment:
You can import the `frame()` helper and use it like this:
```typescript
await frame();
this._syncTrigger();
```
It's just a `Promise` wrapper around `requestAnimationFrame()` for better ergonomics. You'll just need to make the `initialize()` method async which is fine.

When you write tests for this, you just need to await the frame before any assertions to ensure the rAF has time to finish. There should be a lot of examples in existing tests that do this, or use `elementUpdated()` as well.